### PR TITLE
Allow channel urls to be displayed in video's description

### PR DESCRIPTION
**Pull request by ChunkyProgrammer**
closes https://github.com/iv-org/invidious/issues/3751
closes https://github.com/iv-org/invidious/issues/3776

Another thing I noticed that I may tackle in the future is that italic, bold and strikethroughs aren't parsed for the description (at the moment)

### CODE REVIEW

**Comment by SamantazFox** [comment]
I was wondering: we could probably share the code from [`content_to_comment_html`](https://github.com/iv-org/invidious/blob/master/src/invidious/comments.cr#L638-L687) with this one when it comes to url/watch endpoints. If I'm not mistaken, the data uses the same structure.

**Comment by ChunkyProgrammer** [comment]
> I was wondering: we could probably share the code from [`content_to_comment_html`](https://github.com/iv-org/invidious/blob/master/src/invidious/comments.cr#L638-L687) with this one when it comes to url/watch endpoints. If I'm not mistaken, the data uses the same structure.

Good point, did you want me to add the condition from this PR to the content to html? Or did you have something else in mind?

**Comment by SamantazFox** [comment]
I was thinking about a separate `parse_url_endpoint` function or similar

**Comment by ChunkyProgrammer** [comment]
Moved the endpoint logic to a utils function

File: src/invidious/helpers/utils.cr [hunk 1]
```diff
@@ -389,3 +389,56 @@ def reduce_uri(uri : URI | String, max_length : Int32 = 50, suffix : String = "
   end
   return str
 end
+
+# Get the html link from a NavigationEndpoint or an innertubeCommand
+def parse_link_endpont(endpoint : JSON::Any, text : String, video_id : String)
```
Comment (SamantazFox) [inline src/invidious/helpers/utils.cr:394]: You meant `parse_link_endpoint` (with a `i`)? :P

File: src/invidious/comments.cr [hunk 2]
```diff
@@ -635,55 +635,8 @@ def content_to_comment_html(content, video_id : String? = "")
...
-          # See:
-          #  - https://github.com/iv-org/invidious/issues/3038
-          #  - https://github.com/iv-org/invidious/issues/3062
-          text = %(<a href="#{url}">#{text}</a>)
-        else
-          text = %(<a href="#{url}">#{reduce_uri(url)}</a>)
-        end
-      end
+    if navigationEndpoint = run.dig?("navigationEndpoint")
+      text = parse_link_endpont(navigationEndpoint, text, video_id)
```
Comment (ChunkyProgrammer) [inline src/invidious/comments.cr:639]:
```suggestion
      text = parse_link_endpoint(navigationEndpoint, text, video_id)
```

File: src/invidious/helpers/utils.cr (hunk 1 above)
Comment (ChunkyProgrammer) [inline src/invidious/helpers/utils.cr:394]:
```suggestion
def parse_link_endpoint(endpoint : JSON::Any, text : String, video_id : String)
```

File: src/invidious/videos/description.cr [hunk 3]
```diff
@@ -94,7 +49,11 @@ def parse_description(desc : JSON::Any?) : String?
         copy_string(str2, iter, cmd_length)
       end
 
-      str << parse_command(command, cmd_content)
+      link = cmd_content
+      if on_tap = command.dig?("onTap", "innertubeCommand")
+        link = parse_link_endpont(on_tap, cmd_content, video_id)
```
Comment (ChunkyProgrammer) [inline src/invidious/videos/description.cr:54]:
```suggestion
        link = parse_link_endpoint(on_tap, cmd_content, video_id)
```

File: src/invidious/helpers/utils.cr (hunk 1 above)
Comment (ChunkyProgrammer) [reply src/invidious/helpers/utils.cr]: Updated :-)

**Review by SamantazFox — APPROVED** [review]

**Comment by SamantazFox** [comment]
Thanks for contributing ^^

**Pull request merged**